### DISCUSSION TITLE
How to turn LangFlow into an API service?

### DISCUSSION THREAD

**Discussion by ops-004** [langflow-ai/langflow#1136] [Q&A]
Hello, I would like to use LangFlow as an API service instead of a prototyping tool. As a remote API service, I need to address the two most critical issues:
1. Single point of failure problem.
2. Concurrent execution of multiple tasks.

My idea is to deploy multiple LangFlow nodes on multiple physical machines and use Nginx load balancing to solve the single point of failure and concurrent task execution problems.

However, there is still one uncertain point in this solution: After LangFlow-1 node creates a flow or performs other configurations, can LangFlow-2 share these changes?

I have reviewed some information but still cannot find a solution to my doubt. So, I would like to confirm with you whether LangFlow currently supports deploying on multiple machines and nodes to build a highly available distributed API application. This is crucial for me. Do you have any related case studies to share?

Some of my other thoughts and questions:
1. Does the LangFlow API support streaming responses?
2. Can LangFlow export the built flow as LangChain code? This way, I can deploy the code independently as a service using some server tools!

Translation:
Hello, I would like to use LangFlow as an API service instead of it appearing as a prototyping tool.
As a remote API service, I need to address two core issues:
1. Single point of failure problem.
2. Concurrent execution of multiple tasks.

My current idea is to deploy multiple LangFlow nodes on multiple physical machines and utilize Nginx load balancing to address the single point of failure and concurrent task execution problems.

However, this solution still has one uncertain aspect: for example, after making changes such as creating a flow on LangFlow-1 node, would LangFlow-2 be able to share these modifications?

After reviewing some material, I still haven't been able to resolve my doubts, so I wanted to confirm with you whether LangFlow can currently be deployed across multiple machines and nodes to create a highly available, distributed API application. This is extremely important to me. If it is possible, do you have any related case studies?

Some other thoughts and questions I have:
1. Does the LangFlow API support streaming responses?
2. Can LangFlow export the constructed flows as LangChain code? This way, I could independently deploy the code as a service using various server tools!

**Comment by jundaychan** [comment]
Maybe you should take a look **bisheng**

**Comment by zazer0** [comment]
Hi, I have a similar use case; did you end up finding a solution?

**Reply by jundaychan** [reply]
what case?

**Comment by carlosrcoelho** [comment]
Hi! :wave:
We are using other channels as our official means of communication with users. We apologize for the delayed response.
For faster assistance, please direct your questions to our Discord channel or open a GitHub issue, as we are much more active on these platforms.
[Langflow Discord Server](https://discord.com/invite/EqksyE2EX9)
Thank you for your understanding.
